PWGHF/TableProducer/candidateCreatorBs.cxx

```diff
@@ -0,0 +1,453 @@
+          if (arrDaughDsIndex.size() == 2) {
+            for (auto iProng = 0u; iProng < arrDaughDsIndex.size(); ++iProng) {
+              auto daughI = particlesMc.rawIteratorAt(arrDaughDsIndex[iProng]);
+              arrPDGDaughDs[iProng] = std::abs(daughI.pdgCode());
+            }
+            if ((arrPDGDaughDs[0] == arrPDGResonantDsPhiPi[0] && arrPDGDaughDs[1] == arrPDGResonantDsPhiPi[1]) || (arrPDGDaughDs[0] == arrPDGResonantDsPhiPi[1] && arrPDGDaughDs[1] == arrPDGResonantDsPhiPi[0])) {
+              flag = sign * BIT(hf_cand_bs::DecayTypeMc::BsToDsPiToKKPiPi);
+            }
+          }
+        } else {
+          debug = 1;
+          LOGF(debug, "WARNING: Bs decays in the expected final state but the condition on the intermediate state DsPi->PhiPiPi is not fulfilled");
+        }
+      }
+
+      // Partly reconstructed decays, i.e. the 4 prongs have a common b-hadron ancestor
+      // convention: final state particles are prong0,1,2,3
+      if (!flag) {
+        auto particleProng0 = arrayDaughtersBs[0].mcParticle();
+        auto particleProng1 = arrayDaughtersBs[1].mcParticle();
+        auto particleProng2 = arrayDaughtersBs[2].mcParticle();
+        auto particleProng3 = arrayDaughtersBs[3].mcParticle();
+        // b-hadron hypothesis
+        std::array<int, 3> bHadronMotherHypos = {pdg::Code::kB0, pdg::Code::kBS, pdg::Code::kLambdaB0};
+
+        for (const auto& bHadronMotherHypo : bHadronMotherHypos) {
+          int index0Mother = RecoDecay::getMother(particlesMc, particleProng0, bHadronMotherHypo, true);
+          int index1Mother = RecoDecay::getMother(particlesMc, particleProng1, bHadronMotherHypo, true);
+          int index2Mother = RecoDecay::getMother(particlesMc, particleProng2, bHadronMotherHypo, true);
+          int index3Mother = RecoDecay::getMother(particlesMc, particleProng3, bHadronMotherHypo, true);
+
+          // look for common b-hadron ancestor
+          if (index0Mother > -1 && index1Mother > -1 && index2Mother > -1 && index3Mother > -1) {
+            if (index0Mother == index1Mother && index1Mother == index2Mother && index2Mother == index3Mother) {
+              flag = BIT(hf_cand_bs::DecayTypeMc::PartlyRecoDecay);
+              break;
+            }
+          }
+        }
+      }
+
+      rowMcMatchRec(flag, origin, debug);
+    } // rec
+
+    // Match generated particles.
+    for (auto const& particle : particlesMc) {
+      flag = 0;
+      origin = 0;
+      arrDaughDsIndex.clear();
+      // Bs(bar) → Ds∓ π±
+      if (RecoDecay::isMatchedMCGen(particlesMc, particle, pdg::Code::kBS, array{-static_cast<int>(pdg::Code::kDS), +kPiPlus}, true)) {
+        // Match Ds∓ -> K- K+ π±
+        auto candDsMC = particlesMc.rawIteratorAt(particle.daughtersIds().front());
+        if (RecoDecay::isMatchedMCGen(particlesMc, candDsMC, -static_cast<int>(pdg::Code::kDS), array{-kKPlus, +kKPlus, -kPiPlus}, true, &sign, 2)) {
```

Comment:
Since you defined `pdg::Code::kDSBar` you can use it also here
```suggestion
        if (RecoDecay::isMatchedMCGen(particlesMc, candDsMC, pdg::Code::kDSBar, array{-kKPlus, +kKPlus, -kPiPlus}, true, &sign, 2)) {
```